### DISCUSSION TITLE
[OPEN] DHCP Lease not renewed on 2 of 4 identical devices (custom compile)

### DISCUSSION THREAD

**Discussion by lhurt** [arendst/Tasmota#18347] [Support request]
### Problem
There are 4 identical Sonoff iFan3 devices for my ceiling fans. They are [supported by tasmota](https://tasmota.github.io/docs/devices/Sonoff-iFan03/). 2 of them don't do a DHCP lease refresh that causes the devices to become unaccessible as the leases time out and DNS can't resolve the ip anymore.
Only a power cycle can trigger the dhcp lease request again and that's pretty unhandy to climb a ladder and power cycle the device (don't want to flick the fuse as my 24/7 server is on it as well).

### (Maybe?) Helpful additional Information
- Identical 12.4.0 firmware that I compiled by myself to enable MQTT TLS
- 2 devices work as expected and renew their IP every 5 minutes (set in DHCP server)
- all devices are the same hardware, they were actually bought in one batch
- IP Address is set to 0.0.0.0 on all devices. They request their IP on reboot as expected

### Actions tried for resolution
- Updated to a new version. The issue was already present with a previous 12.3.x version
- Reset the device with `reset 3` and `reset 1` and did powercycle the device after
- Tried official firmware
- Reflashed one of the not working devices from scratch without and also configuration from scratch

### Known workarounds
- reboot the device every day via API (that's dirty)
- assign a fixed IP, but I don't want to mess around with IPs. I'm planning to change my base network and this would be nightmare

Any ideas what happens or what I do wrong or what I could do to get dhcp going?
Thanks a lot in advance.

**Comment by schrej-zz** [comment]
Hi,
I have the same issue now.
I changed the subnet mask of the DHCP server (Fritz!Box) and afterwards the devices had an IP within the open DHCP range despite I assigned a dedicated IP address for them in DHCP.
Even Restart 1 didn't change the IP address to the assigned one.

**Reply by s-hadinger** [reply]
This is a router issue. Tasmota does not remember IP addresses after a reboot.

**Reply by schrej-zz** [reply]
Hi,
I can confirm this.
The Fritz!Box lost the fixed assignements to IP an address after subnetmask change.
Thanks
